File: Part8/C#/FunctionConcepts/FunctionConcepts_7.cs
// Example of function overloading
using System;

class Program
{
    static void MyFunction ( int IntParameter_Par )
    {
        Console.WriteLine ( "Int version of my function called {0}", IntParameter_Par );
    }

    static void MyFunction ( double DoubleParameter_Par )
    {
        Console.WriteLine ( "Double version of my function called {0:F1}", DoubleParameter_Par );
    }

    static void Main ()
    {
        MyFunction ( IntParameter_Par: 5 );
        MyFunction ( DoubleParameter_Par: 5.5 );
    }
}

File: Part8/Python/FunctionConcepts/FunctionConcepts_7.py
# Example of function overloading
# Python does not support traditional function overloading, last definition wins
def myFunction ( IntParameter_Par ):
    print ( f"Int version of my function called {IntParameter_Par}" )

def myFunction ( DoubleParameter_Par ):
    print ( f"Double version of my function called {DoubleParameter_Par}" )

def main():
    myFunction ( 5 )
    myFunction ( 5.5 )

if __name__ == "__main__":
    main()
